homeassistant/components/overkiz/water_heater_entities/atlantic_dhw.py

```diff
@@ -0,0 +1,175 @@
+        self._attr_min_temp = cast(
+            float,
+            self.executor.select_state(
+                OverkizState.CORE_MINIMAL_TEMPERATURE_MANUAL_MODE
+            ),
+        )
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return cast(
+            float,
+            self.executor.select_state(
+                OverkizState.MODBUSLINK_MIDDLE_WATER_TEMPERATURE
+            ),
+        )
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature corresponding to the PRESET."""
+        return cast(
+            float,
+            self.executor.select_state(OverkizState.CORE_WATER_TARGET_TEMPERATURE),
+        )
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new temperature."""
+        temperature = kwargs[ATTR_TEMPERATURE]
+        await self.executor.async_execute_command(
+            OverkizCommand.SET_TARGET_DHW_TEMPERATURE, temperature
+        )
+
+    @property
+    def is_boost_mode_on(self) -> bool:
+        """Return true if boost mode is on."""
+        return self.executor.select_state(OverkizState.MODBUSLINK_DHW_BOOST_MODE) in (
+            OverkizCommandParam.ON,
+            OverkizCommandParam.PROG,
+        )
+
+    @property
+    def is_eco_mode_on(self) -> bool:
+        """Return true if eco mode is on."""
+        return self.executor.select_state(OverkizState.MODBUSLINK_DHW_MODE) in (
+            OverkizCommandParam.MANUAL_ECO_ACTIVE,
+            OverkizCommandParam.AUTO_MODE,
+        )
+
+    @property
+    def is_away_mode_on(self) -> bool:
+        """Return true if away mode is on."""
+        return (
+            self.executor.select_state(OverkizState.MODBUSLINK_DHW_ABSENCE_MODE)
+            == OverkizCommandParam.ON
+        )
+
+    @property
+    def current_operation(self) -> str:
+        """Return current operation."""
+        if self.is_away_mode_on:
+            return STATE_OFF
+
+        if self.is_boost_mode_on:
+            return STATE_PERFORMANCE
+
+        if self.is_eco_mode_on:
+            return STATE_ECO
+
+        if (
+            cast(str, self.executor.select_state(OverkizState.MODBUSLINK_DHW_MODE))
+            == OverkizCommandParam.MANUAL_ECO_INACTIVE
+        ):
+            return OverkizCommandParam.MANUAL
+
+        return STATE_OFF
+
+    async def async_set_operation_mode(self, operation_mode: str) -> None:
+        """Set new operation mode."""
+        if operation_mode in (STATE_PERFORMANCE, OverkizCommandParam.BOOST):
+            await self.async_turn_away_mode_off()
+            await self.async_turn_boost_mode_on()
+        elif operation_mode in (
+            OverkizCommandParam.ECO,
+            OverkizCommandParam.MANUAL_ECO_ACTIVE,
+        ):
+            await self.async_turn_away_mode_off()
+            await self.async_turn_boost_mode_off()
+            await self.executor.async_execute_command(
```

Comment:
Don't you face issues with too many commands in a short time? A potential fix could be to first check the state before turning off away and boost mode. No need to call the command, if this is already off.

Reply:
I did this by an example in a neighboring device on this integration, so I didn't test it.
Yes, the API randomly gives error 429, but the requests get re-sent by the integration.
I guess it would be better to rate-limit the requests in the integration core, but ok, I will try to play with this to save two of these calls.

Reply:
done. added the checks